src/apme_engine/daemon/primary_server.py

```
@@ -1702,6 +1764,257 @@ async def async_scan_fn(file_paths: list[str]) -> list[ViolationDict]:
+        Args:
+            remediate_task: Running graph remediation task.
+        """
+        remediate_task.cancel()
+        try:
+            await remediate_task
+        except asyncio.CancelledError:
+            pass
+        except Exception:
+            logger.exception("remediate_task cleanup failed after deadline enforcement")
+
+    async def _drain_remediate_task(  # noqa: DOC404
```

Ruff (0.16.0)

[warning] Invalid rule code in `# noqa`: DOC404

Add non-Ruff rule codes to the `lint.external` configuration option

(RUF102)